src/main/java/de/tum/in/www1/artemis/service/ExampleSubmissionService.java

```diff
@@ -73,7 +73,7 @@ public ExampleSubmission save(ExampleSubmission exampleSubmission) {
         Result result = submission.getResult();
 
         // result.isExampleResult() can have 3 values: null, false, true. We return if it is not true
-        if (result == null || result.isExampleResult() != Boolean.TRUE) {
+        if (result == null || !Boolean.TRUE.equals(result.isExampleResult())) {
```

Comment:
I see why we would keep this as != true (because it's possible we don't understand why the original programmer chose to do it this way, and we don't want to have unintended consequences), but I'm wondering if Boolean.FALSE.equals would be better. Minor detail and to me it doesn't require changes.

Reply:
Keep in mind that Boolean has 3 states. False, True and null

Reply:
The reason why I chose to do it that way is that `Boolean.FALSE.equals(null)` is `false` but `null != Boolean.TRUE` and `!Boolean.TRUE.equals(null)` are both `true`. This is one of the pitfalls when dealing with the wrapper classes. But I do agree that the other variant is better readable.

Maybe we later need to look if that can be simplified or converted, it is however a change in semantics.